roles/rsyslog/roles/output_roles/files/defaults/main.yml

```diff
@@ -34,72 +34,49 @@ __rsyslog_conf_files_output_rules:
 
   - name: 'output-files'
     type: 'output'
-    suffix: 'system'
-    path: '{{ rsyslog_config_dir }}'
     sections:
 
       - options: |-
-          {% if rsyslog_files_actions | d([]) %}
-          {% for element in rsyslog_files_actions %}
-          {% if element.name is defined and element.path is defined %}
-          {% if element.severity is defined %}
-          {% if element.facility is defined %}
+          {% if rsyslog_output_files | d([]) and rsyslog_output_files.0.path | d() %}
```

Comment:
Is the logic here `if rsyslog_output_files has some elements, and the first element has the path attribute, output the configuration for each element, else output the default configuration`?  Does that mean it is possible for `rsyslog_output_files` to contain 1 element that has no `path` attribute?

Because you could get rid of one level of `if` by doing something like this: https://jinja.palletsprojects.com/en/2.11.x/templates/#for
```
{% for element in rsyslog_output_files | d([]) %}
... output config for each element ...
{% else %}
.... output default config ...
{% endfor %}
```
if you need to loop only for those elements that have a `path` attribute:
```
{% for element in rsyslog_output_files | d([]) | selectattr("path","defined") %}
... output config for each element that has a "path" attribute ...
{% else %}
.... output default config when no list elements have a "path" attribute ...
{% endfor %}
```

Reply:
> Is the logic here `if rsyslog_output_files has some elements, and the first element has the path attribute, output the configuration for each element, else output the default configuration`? Does that mean it is possible for `rsyslog_output_files` to contain 1 element that has no `path` attribute?
> ...

Well, I do not know from system roles perspective, but from rsyslog side file output with no path does not make sense and is actually not allowed (you will get an error in log complaining about it, if you write it in config).

Reply:
This logic - check the first rsyslog_output_files item to have the path or not - supports this simplest configuration.  The files output - writing to the local files - is implicitly configured.  
```
        logging_enabled: true
        logging_inputs:
          - name: basic_input
            type: basics
```
The following input code @richm suggested me to enhance is the exact case.  Here, the files output is not configured, and we add it without a path parameter.  Then, in the files output code, it goes to {% else %} and the default actions - rsyslog.conf in the rpm package - are deployed.
```
          {% if not rsyslog_outputs|selectattr("type","==","files")|list %}
          {%   set rsyslog_outputs = rsyslog_outputs + [{"name": "files", "type": "files"}] %}
          {% endif %}
```